lib/beaker/ssh_connection.rb

```diff
@@ -147,12 +149,16 @@ def wait_for_connection_failure options = {}, stdout_callback = nil, stderr_call
         slept = 0
         stdout_callback.call("sleep #{wait} second(s): ")
         while slept < wait
-          sleep slept
+          sleep 1
           stdout_callback.call('.')
           slept += 1
         end
         stdout_callback.call("\n")
-        (last_wait, wait) = wait, last_wait + wait
+        if poll_interval
+          wait = poll_interval
+        else
+          (last_wait, wait) = wait, last_wait + wait
+        end
```

Comment:
The motivation is that I am doing snapshot and restore VMs for the HA tests and using this method to determine when the SSH connection goes down - meaning that the restore has begun, then waiting for the connection to come back up afterwards.  The problem was that if it went through a few iterations of waiting for the connection failure then it would sometimes completely miss the agent going offline since it goes offline for only 15 seconds or so and then get stuck waiting forever for the connection to fail.  I am thinking the same issue could arise if for example someone were using this method to wait for a system reboot to happen - although this would be less likely because the system reboot usually begins nearly immediately and can take quite a while to complete.